contributors/design-proposals/scheduling/taint-node-by-condition.md

```diff
@@ -15,26 +15,32 @@ Currently (1.8 and before), the conditions of nodes are updated by the kubelet a
-|                    |False               | -            | |
-|                    |Unknown             | -            | |
-|MemoryPressure      |True                | NoSchedule   | node.kubernetes.io/memory-pressure     |
-|                    |False               | -            | |
-|                    |Unknown             | -            | |
-|DiskPressure        |True                | NoSchedule   | node.kubernetes.io/disk-pressure       |
-|                    |False               | -            | |
-|                    |Unknown             | -            | |
-|NetworkUnavailable  |True                | NoSchedule   | node.kubernetes.io/network-unavailable |
-|                    |False               | -            | |
-|                    |Unknown             | -            | |
-|PIDPressure         |True                | NoSchedule   | node.kubernetes.io/pid-pressure        |
-|                    |False               | -            | |
-|                    |Unknown             | -            | |
+| ConditionType        | Condition Status   |Effect        | Key      |
+| -------------------- | ------------------ | ------------ | -------- |
+|Ready                 |True                | -            | |
+|                      |False               | NoExecute    | node.kubernetes.io/not-ready                   |
+|                      |Unknown             | NoExecute    | node.kubernetes.io/unreachable                 |
+|OutOfDisk             |True                | NoSchedule   | node.kubernetes.io/out-of-disk                 |
+|                      |False               | -            | |
+|                      |Unknown             | -            | |
+|MemoryPressure        |True                | NoSchedule   | node.kubernetes.io/memory-pressure             |
+|                      |False               | -            | |
+|                      |Unknown             | -            | |
+|DiskPressure          |True                | NoSchedule   | node.kubernetes.io/disk-pressure               |
+|                      |False               | -            | |
+|                      |Unknown             | -            | |
+|NetworkUnavailable    |True                | NoSchedule   | node.kubernetes.io/network-unavailable         |
+|                      |False               | -            | |
+|                      |Unknown             | -            | |
+|PIDPressure           |True                | NoSchedule   | node.kubernetes.io/pid-pressure                |
+|                      |False               | -            | |
+|                      |Unknown             | -            | |
+|NodeShutDown          |True                | NoSchedule   | node.cloudprovider.kubernetes.io/shutdown      |
+|                      |False               |-             | |
+|                      |Unknown             |-             | |
+|ExternalCloudProvider |True                | NoSchedule   | node.cloudprovider.kubernetes.io/uninitialized |
```

Comment:
AFAIK, the `ExternalCloudProvider` is not really a node condition. Kubelet taints a node with this taint when the cloud provider is set to external. So, I guess we should explain here that the node is tainted when the cloud provider is external and leave the `Condition Status` column empty.

Reply:
@bsalamat I have removed condition column from table, also I have added little explanation about taint when it is set.